sdk/python/kfp/_auth.py

```diff
@@ -117,3 +121,32 @@ def get_google_open_id_connect_token(service_account_credentials):
     token_response = google.oauth2._client._token_endpoint_request(
         request, OAUTH_TOKEN_URI, body)
     return token_response['id_token']
+
+def get_id_token(client_id, client_secret, audience):
+    """Obtain the ID token for provided Client ID with user accounts.
+        Flow: get authorization code -> exchange for refresh token -> obtain and return ID token
+    """
+    auth_code = get_auth_code(client_id)
+    refresh_token = get_refresh_token_from_code(auth_code, client_id, client_secret)
+    return id_token_from_refresh_token(client_id, client_secret, refresh_token, audience)
+
+def get_auth_code(client_id):
+    auth_url = "https://accounts.google.com/o/oauth2/v2/auth?client_id=%s&response_type=code&scope=openid%%20email&access_type=offline&redirect_uri=urn:ietf:wg:oauth:2.0:oob"%client_id
```

Comment:
Since we have many authentication -related functions, it might be great to indicate which ones are GCP-specific and which ones are generic. WDYT?

Reply:
SG. Currently, all functions in auth file is google specific. We can later rename it as gcp_auth.py